tdxctl/src/fde_setup.rs

```diff
@@ -181,44 +184,111 @@ impl SetupFdeArgs {
+            .remote_uri(kms_url.clone())
+            .tls_client_cert(cert_pair.cert_pem)
+            .tls_client_key(cert_pair.key_pem)
+            .tls_ca_cert(tmp_ca.ca_cert.clone())
+            .cert_validator(Box::new(|cert| {
+                let Some(cert) = cert else {
+                    bail!("Missing server cert");
+                };
+                let Some(usage) = cert.special_usage else {
+                    bail!("Missing server cert usage");
+                };
+                if usage != "kms:rpc" {
+                    bail!("Invalid server cert usage: {usage}");
+                }
+                Ok(())
+            }))
+            .build()
+            .into_client()
+            .context("Failed to create client")?;
+        let kms_client = kms_rpc::kms_client::KmsClient::new(ra_client);
+        let response = kms_client
+            .get_app_key(GetAppKeyRequest {
+                app_compose: fs::read_to_string(host_shared.dir.app_compose_file())
+                    .context("Failed to read app compose file")?,
+            })
+            .await
+            .context("Failed to get app key")?;
+        {
+            let (_, ca_pem) = x509_parser::pem::parse_x509_pem(tmp_ca.ca_cert.as_bytes())
+                .context("Failed to parse ca cert")?;
+            let x509 = ca_pem.parse_x509().context("Failed to parse ca cert")?;
+            let id = hex::encode(x509.public_key().raw);
+            let provider_info = KeyProviderInfo::new("kms".into(), id);
+            emit_key_provider_info(&provider_info)?;
+        };
+        let keys = AppKeys {
+            ca_cert: response.ca_cert,
+            disk_crypt_key: response.disk_crypt_key,
+            env_crypt_key: response.env_crypt_key,
+            k256_key: response.k256_key,
+            k256_signature: response.k256_signature,
+            tproxy_app_id: response.tproxy_app_id,
+            key_provider: KeyProvider::Kms { url: kms_url },
+        };
+        let keys_json = serde_json::to_string(&keys).context("Failed to serialize app keys")?;
+        fs::write(self.app_keys_file(), keys_json).context("Failed to write app keys")?;
+        Ok(())
+    }
+
+    async fn get_keys_from_local_key_provider(&self, host: &HostApi) -> Result<()> {
+        info!("Getting keys from local key provider");
+        let provision = host
+            .get_sealing_key()
+            .await
+            .context("Failed to get sealing key")?;
+        // write to fs
+        let app_keys =
+            gen_app_keys_from_seed(&provision.sk).context("Failed to generate app keys")?;
+        let keys_json = serde_json::to_string(&app_keys).context("Failed to serialize app keys")?;
+        fs::write(self.app_keys_file(), keys_json).context("Failed to write app keys")?;
+
+        // write to RTMR
+        let provider_info = KeyProviderInfo::new("local-sgx".into(), hex::encode(provision.mr));
+        emit_key_provider_info(&provider_info)?;
```

Comment:
Review Hint: For local key provider, the sgx `mr_enclave` is written to RTMR3. The sgx quote is verified in fn `get_sealing_key`.